lib/puppet/transaction/report.rb

```diff
@@ -176,7 +178,7 @@ def add_resource_status(status)
 
   # @api private
   def compute_status(resource_metrics, change_metric)
-    if resources_failed_to_generate || (resource_metrics["failed"] || 0) > 0
+    if resources_failed_to_generate || (resource_metrics["failed"] || 0) > 0 || @logs.any? { |lg| lg.level == :err }
```

Comment:
Ouch, we still allow logs after finalizing a report?  That seems... wrong.

I would expect that once finalize is called the report becomes immutable - maybe something to file a ticket in the `Improve Agent Lifecycle` epic to fix in the future.